### PR TITLE
`PatternSearchTest` — SQL to AR/Arel

### PR DESCRIPTION
**Pull request by nimmolo**
This PR is small, to facilitate read-throughs.

Next PR will add some new scopes to` Name` and `Observation` and implement the scopes in this and the `API2` tests

### CODE REVIEW

File: test/models/pattern_search_test.rb [hunk 1]
```diff
@@ -229,7 +230,8 @@ def test_parse_list_of_locations
     x.vals = ids.map(&:to_s)
     assert_equal(ids, x.parse_list_of_locations)
     x.vals = ["*California, USA"]
-    expect = Location.where("name LIKE '%California, USA'").map(&:id).sort
+    expect = Location.where(Location[:name].matches("%California, USA")).
+             map(&:id).sort
```
Comment (JoeCohen) [inline test/models/pattern_search_test.rb:234]:
```suggestion
    expect = Location.where(Location[:name].matches("%California, USA")).
             order(id: :asc).pluck(:id)
```

File: test/models/pattern_search_test.rb [hunk 2]
```diff
@@ -208,7 +208,8 @@ def test_parse_list_of_herbaria
     x.vals = ids.map(&:to_s)
     assert_equal(ids, x.parse_list_of_herbaria)
     x.vals = ["*Herbarium"]
-    expect = Herbarium.where("name LIKE '%Herbarium'").map(&:id).sort
+    expect = Herbarium.where(Herbarium[:name].matches("%Herbarium")).
+             map(&:id).sort
```
Comment (JoeCohen) [inline test/models/pattern_search_test.rb:212]:
```suggestion
    expect = Herbarium.where(Herbarium[:name].matches("%Herbarium")).
             order(id: :asc).pluck(:id)
```

File: test/models/pattern_search_test.rb [hunk 3]
```diff
@@ -268,7 +270,8 @@ def test_parse_list_of_species_lists
     x.vals = ids.map(&:to_s)
     assert_equal(ids, x.parse_list_of_species_lists)
     x.vals = ["query*"]
-    expect = SpeciesList.where("title LIKE 'query%'").map(&:id).sort
+    expect = SpeciesList.where(SpeciesList[:title].matches("query%")).
+             map(&:id).sort
```
Comment (JoeCohen) [inline test/models/pattern_search_test.rb:274]:
```suggestion
    expect = SpeciesList.where(SpeciesList[:title].matches("query%")).
             order(id: :asc).pluck(:id)
```

**Review by JoeCohen — COMMENTED** [review]
I like it.

File: test/models/pattern_search_test.rb [hunk 4]
```diff
@@ -470,21 +473,21 @@ def test_observation_search
   end
 
   def test_observation_search_date
-    expect = Observation.where("YEAR(`when`) = 2006")
+    expect = Observation.where(Observation[:when].year == 2006)
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply test/models/pattern_search_test.rb:476]: Ah. Why not just use `year`?

File: test/models/pattern_search_test.rb [hunk 5]
```diff
@@ -585,14 +589,14 @@ def test_observation_search_gps
   end
 
   def test_observation_search_images_no
-    expect = Observation.where("thumb_image_id IS NULL")
+    expect = Observation.where(Observation[:thumb_image_id] = nil)
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply test/models/pattern_search_test.rb]: Done! Thanks

**Comment by JoeCohen** [comment]
map vs pluck
I didn't realize that it had been benchmarked
I don't know where's the better place to sort: the db or Rails.

**Comment by JoeCohen** [comment]
You're right. I couldn't figure out why it was failing, so my "improvement"
was a failure.

File: test/models/pattern_search_test.rb [hunk 6]
```diff
@@ -627,21 +631,21 @@ def test_observation_search_has_names_no
   end
 
   def test_observation_search_has_names_yes
-    expect = Observation.where("name_id != #{names(:fungi).id}")
+    expect = Observation.where(Observation[:name_id] != names(:fungi).id)
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply test/models/pattern_search_test.rb:634]: done. good catch

File: test/models/pattern_search_test.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (nimmolo) [reply test/models/pattern_search_test.rb]: This is much more concise. Thanks

**Comment by nimmolo** [comment]
There will be another chance to refine many of these (in one easier-to-read place) in a next PR, where i'll put these as scopes in the models, and implement the scopes in the tests — the API and controller tests make a lot of these same queries. 

In this PR, I just wanted to get the "query-expect" statements working in AR/Arel as a kind of first draft of the scopes.

**Comment by nimmolo** [comment]
@JoeCohen — I played around with the SQL generated by `Observation.no_notes_persisted` in the console, and finally realized there is already a method that generates the exact same SQL as the current method, and does not require SQL interpolation or disabling the cop: `Observation.no_notes`. I have switched to that here.

**Comment by nimmolo** [comment]
I believe this is good to go. Further changes to `PatternSearchTest` ahead in the scopes PR, #1082.
Would like to merge if no objections.

**Pull request merged**